docs/development.rst

```diff
@@ -64,119 +124,581 @@ Quickstart
-    metric was added to 
+    metric was added to
     `test_topology.py <https://github.com/tskit-dev/tskit/blob/master/python/tests/test_topology.py>`_.
     Add a python version of your function here.
 3.  Create a new class in this module to write unit tests for your function: in addition
     to making sure that your function is correct, make sure it fails on inappropriate inputs.
-    This can often require judgement. For instance, :meth:`Tree.kc_distance` fails on a tree 
+    This can often require judgement. For instance, :meth:`Tree.kc_distance` fails on a tree
     with multiple roots, but allows users to input parameter values that are nonsensical,
-    as long as they don't break functionality. 
+    as long as they don't break functionality.
 4.  Write your function in C: check out the :ref:`sec_c_api` for guidance. There
-    are also many examples in the 
+    are also many examples in the
     `c directory <https://github.com/tskit-dev/tskit/tree/master/c/tskit>`_.
-    Your function will probably go in 
+    Your function will probably go in
     `trees.c <https://github.com/tskit-dev/tskit/blob/master/c/tskit/trees.c>`_.
     For C debugging you can force a debug build of the `_tskit module with
     `CFLAGS='-Wall -O0 -g' make` in the `python` directory.
-5.  Write a few tests for your function in C: again, write your tests in  
+5.  Write a few tests for your function in C: again, write your tests in
     `tskit/c/tests/test_tree.c <https://github.com/tskit-dev/tskit/blob/master/c/tests/test_trees.c>`_.
     The key here is code coverage, you don't need to worry as much about covering every
     corner case, as we will proceed to link this function to the Python tests you
     wrote earlier.
 6.  Create a low-level definition of your function using Python's C API: this will
-    go in `_tskitmodule.c 
+    go in `_tskitmodule.c
     <https://github.com/tskit-dev/tskit/blob/master/python/_tskitmodule.c>`_.
 7.  Test your low-level implementation in `tskit/python/tests/test_lowlevel.py
     <https://github.com/tskit-dev/tskit/blob/master/python/tests/test_lowlevel.py>`_:
     again, these tests don't need to be as comprehensive as your first python tests,
     instead, they should focus on the interface, e.g., does the function behave
     correctly on malformed inputs?
-8.  Link your C funtion to the Python API: write a function in tskit's Python API, 
-    for example the kc_distance function lives in 
-    `tskit/python/tskit/trees.py 
+8.  Link your C function to the Python API: write a function in tskit's Python API,
+    for example the kc_distance function lives in
+    `tskit/python/tskit/trees.py
     <https://github.com/tskit-dev/tskit/blob/master/python/tskit/trees.py>`_.
 9.  Modify your Python tests to test the new C-linked function: if you followed
     the example of other tests, you might need to only add a single line of code
     here.
```

Comment:
What happens to the Python version at this point - maybe, we should say to write new tests that (non-comprehensively) compare the python version to the C version?